File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int open = 0, max = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                open++;
                if (open > max) max = open;
            } else if (s.charAt(i) == ')') open--;
        }
        return max;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int res = 0;
        
        for (int i = 0, n = s.size(), open = 0; i < n; i++) {
            if (s[i] == '(') {
                open++;
                res = max(res, open);
            } else if (s[i] == ')') {
                open--;
            }
        }
        
        return res;
    }
};